packages/core/src/MatchMaker.ts

```diff
@@ -305,17 +305,21 @@ export async function createRoom(roomName: string, clientOptions: ClientOptions)
   }
 
   if (isDevMode) {
-    presence.hset(getRoomRestoreListKey(), room.roomId, JSON.stringify({
-      "clientOptions": clientOptions,
-      "roomName": roomName,
-      "processId": processId
+    await presence.hset(getRoomRestoreListKey(), room.roomId, JSON.stringify({
```

Comment:
awaiting here is likely to cause timing changes that could break things. Currently, the key shows up sometime after createRoom has returned. awaiting here will cause it to show up first, which means more requests to join the room come in before createRoom has even returned.